amethyst_tiles/src/morton/mod.rs

```diff
@@ -79,11 +81,23 @@ pub fn morton_decode_intr_3d(morton: u32) -> (u32, u32, u32) {
     )
 }
 
+fn next_power_of_2(mut v: u32) -> u32 {
```

Comment:
whoa what is this magic. Let's see what happens when `v` begins as 32:

```rust
// using base 2
// v: 0010 0000

v -= 1;       // 0001 1111

              // 0001 1111 bitwise OR
              // 0000 1111
v |= v >> 1;  // 0001 1111

              // 0001 1111 bitwise OR
              // 0000 0111
v |= v >> 2;  // 0001 1111

              // 0001 1111 bitwise OR
              // 0000 0001
v |= v >> 4;  // 0001 1111
v |= v >> 8;  // ..
v |= v >> 16; // ..

v + 1         // 0010 0000
```

hm, this doesn't seem right. did I do something wrong? what's it meant to do?

Reply:
This causes the dimensions to round UP to the next bitwise 2^n. Because Morton encoding is bit interleaving, we need to work within these bounds for indexing. So for example, it would round 20,20 up to 32,32.

Your example is correct, because it is a 2^n already, so stays at 32.

Reply:
There is a builtin for that. https://doc.rust-lang.org/std/primitive.u32.html#method.next_power_of_two